frontend/src/store/wallet.ts

```diff
@@ -219,18 +278,23 @@ export default function useWalletStore() {
    * @notice Prompts user for a signature to change network
    */
   async function setNetwork(chain: Chain) {
+    setLoading(true);
+
     try {
       await provider.value?.send('wallet_switchEthereumChain', [{ chainId: chain.chainId }]);
     } catch (error) {
       // This error code indicates that the chain has not been added to MetaMask.
-      if (error.code === 4902) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if (((error as never) as any).code === 4902) {
```

Comment:
Why do we need this new casting?

Reply:
Maybe better to do `} catch (error as {code: string}) {`

Reply:
New casting is needed because of `Object is of type 'unknown'.ts(2571)`

`error as {code: string}` This doesn't work because of `Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)` https://github.com/microsoft/TypeScript/issues/20024.

I fixed this casting like this:
```
 catch (switchError) {
      const { code } = switchError as { code: number };
      if (code === 4902) {}
}
```